### ISSUE TITLE
[BUG] M306 T (MPC Autotune) does not use part cooling fan when ANYCUBIC_KOSSEL_ENABLE_BED is set to 0

### ISSUE THREAD

**Issue by CrashTD** [MarlinFirmware/Marlin#24839]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

Set `#define ANYCUBIC_KOSSEL_ENABLE_BED` to **0** (**2** is default) leads to the part cooling fan not running during cooling and heat loss testing phase for MPC Autotune via `M306 T`

I'm using a Anycubic Kossel Linear Plus printer in stock configuration (-hotbed, +3DTouch probe) with the example config: https://github.com/MarlinFirmware/Configurations/tree/bugfix-2.1.x/config/examples/delta/Anycubic/Kossel%20Linear%20Plus

Changes made:

```diff
- #define ANYCUBIC_KOSSEL_ENABLE_BED 2
+ #define ANYCUBIC_KOSSEL_ENABLE_BED 0
```

```diff
- #define PIDTEMP          // See the PID Tuning Guide at https://reprap.org/wiki/PID_Tuning
- //#define MPCTEMP        // ** EXPERIMENTAL **
+ //#define PIDTEMP          // See the PID Tuning Guide at https://reprap.org/wiki/PID_Tuning
+ #define MPCTEMP        // ** EXPERIMENTAL **
```

---

To prevent nozzle crashing on bed
```diff
-  #define DELTA_HEIGHT 300.00             // (mm) Get this value from G33 auto calibrate
+  #define DELTA_HEIGHT 290.75             // (mm) Get this value from G33 auto calibrate
```

```diff
-#define ANYCUBIC_PROBE_VERSION 2
+#define ANYCUBIC_PROBE_VERSION 0
```

^ this two changes make no difference except making a new dent in my printbed when not done...
cooling fan behaves the same even if not changed.

### Expected behavior

Cooling fan working during "cooling.." and "testing heat loss" phases even when there is no heatbed installed.

### Actual behavior

Cooling phase can take very long if hotend was still warm since marlin seems to wait for low enough temperature which hotend now has to reach through radiation alone..

Heat loss testing phase not giving reliable data since influence through cooling fan is missing. Results in fast temperature drop as soon cooling fans start after printing first layer, causing thermal runaway error and consequently failed print.

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

Anycubic Kossel Linear Plus

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

Cura

### Host Software

Same as my slicer

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/9728897/Configuration.zip)

---

**Edit**

The reason for this behaviour seems to be, that the Index for the cooling fan changes.

With `#define ANYCUBIC_KOSSEL_ENABLE_BED 2` the cooling fan can be correctly controlled with `M106 P0 S255` and `M306 T` works as expected.

With `#define ANYCUBIC_KOSSEL_ENABLE_BED 0` however, the index of the cooling fan changes to 1. `M106 P0 S255` does nothing, `but M106 P1 S255` brings it to full speed. However, the latter is not used by `M306 T` but it still tries to address index 0.

**Comment by tombrazier** [comment]
After a discussion and some testing with @CrashTD it appears that with `#define ANYCUBIC_KOSSEL_ENABLE_BED 0` the ~~fan cooling the hotend~~ part cooling fan for the hotend changes from fan 0 to fan 1. I think this is a consequence of the PWM FET ordering logic in `Marlin/src/pins/ramps/pins_TRIGORILLA_14.h`.

Whether this is a bug in the config or an appropriate remapping of output MOSFETs in the absence of a bed output, I will have to leave to someone who knows more about the Trigorilla mainboard in question.

**Comment by CrashTD** [comment]
Just for clarification, it is not the hotend fan that is changing but the part cooling fan (aka. filament fan). the hotend fan starts automatic as soon the temperatur is >50°C and works as expected in any configuration.

i will try to dig a bit deeper in this behaviour and do some more test, but since i'm not a dev and have zero experience it would of course help if someone could give direction or ask targeted questions.

**Comment by tombrazier** [comment]
Good point. I have corrected my previous comment.

The thing to check is what MOSFET actually switches on when `M106 P0` is used with no bed. My guess is it is the one that would otherwise be driving the bed. This is now the output for part cooling fan 0. And the output that formerly drove fan 0 now drives fan 1. If the board is marked in a way that contradicts this arrangement, then the config file I referenced above should, perhaps, be changed.
